### PR TITLE
Remaining accessibility fixes to the PL site

### PR DESCRIPTION
**Pull request by clrux**
This relatively small bit of work cleans up the remaining (pertinent) accessibility issues.

Some work performed:

* contrast issues on links, especially those on gray
* set background and foregrounds on the `.sr-only` elements (to black and white, respectively)
* added a few `aria-label`'s on landmark elements
* markup and structure improvements

Some errors that aren't relevant:

* form field examples don't have submit buttons; this is okay since these are just examples
* a few of the colors have contrast issues; these are intentionally shown and we note them visually

---

Please review:

- [x] @talbs 
- [x] @cptvitamin

### CODE REVIEW

**Comment by clrux** [comment]
@talbs The grid boxes don't provide enough contrast. The gray on pink needs a bit more. Do you want to play around with the colors? For what it's worth, I'm okay leaving it as is, so long as we provide textual equivalents for the classnames. But this is mostly visual and so I don't think it's a showstopper.

That is the only outstanding issue here. Let me know.

File: _includes/navigation.html [hunk 1]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
-    <h2 class="pldoc-heading pldoc-nav-title hd-6">Documentation</h2>
-    <ul class="pldoc-list nav-list list-styleguides">
+<nav class="pldoc-nav pldoc-nav-documentation pldoc-nav-external" id="pldoc-nav-documentation" role="navigation" aria-label="Documentation">
+    <h2 class="pldoc-heading pldoc-nav-title hd-6" id="heading-documentation">Documentation</h2>
+    <ul class="pldoc-list nav-list list-styleguides" aria-labelledby="heading-documentation">
         <li class="nav-list-item">
             <a class="pldoc-link" rel="external" href="https://github.com/edx/ux-pattern-library/wiki/Styleguide:-General">
                 <span class="pldoc-icon" aria-hidden="true">&#65514;</span>
                 General Styleguide
+                <span class="sr-only">Opens in a new window</span>
```
Comment (talbs) [inline _includes/navigation.html]: Please add parenthesis to these.

File: _posts/elements/2015-05-12-forms.md [hunk 2]
```diff
@@ -53,22 +53,22 @@ info: Forms allow users to interact with the interface, set preferences, and ent
                 </div>
             </div>
             <div class="field">
-                <label class="field-label field-required" for="email_address_default_suc_helper">Email address (default, has success)</label>
-                <input class="field-input input-text has-success" type="email" id="email_address_default_suc_helper" name="email_address_default_suc_helper" placeholder="e.g. jane@email.com" value="jane@domain.com" aria-describedby="email_address_default_suc_helper" required>
-                <div class="field-hint" id="email_address_default_suc_helper">
+                <label class="field-label field-required" for="email-address-default-suc-helper">Email address (default, has success)</label>
+                <input class="field-input input-text has-success" type="email" id="email-address-default-suc-helper" name="email-address-default-suc-hint" placeholder="e.g. jane@email.com" value="jane@domain.com" aria-describedby="email-address-default-suc-hint" required>
```
Comment (talbs) [inline _posts/elements/2015-05-12-forms.md]: Please do not abbreviate ``suc``? Let's match the ``success`` label/term used elsewhere in the DOM.

File: _posts/elements/2015-05-12-forms.md [hunk 3]
```diff
@@ -23,29 +23,29 @@ info: Forms allow users to interact with the interface, set preferences, and ent
...
+                <div class="field-hint" id="email-address-default-hint">
                     <p>Your email address will never be shared. We promise.</p>
                 </div>
             </div>
             <div class="field">
-                <label class="field-label field-required" for="email_address_default_err_helper">Email address (default, has error)</label>
-                <input class="field-input input-text has-error" type="email" id="email_address_default_err_helper" name="email_address_default_err_helper" placeholder="e.g. jane@email.com" value="jane@domain" aria-describedby="email_address_default_err_helper" required>
+                <label class="field-label field-required" for="email-address-default-err-hint">Email address (default, has error)</label>
+                <input class="field-input input-text has-error" type="email" id="email-address-default-err-hint" name="email-address-default-err-hint" placeholder="e.g. jane@email.com" value="jane@domain" required>
```
Comment (talbs) [inline _posts/elements/2015-05-12-forms.md]: Please do not abbreviate ``err``? Let's match the ``error`` label/term used elsewhere in the DOM.

**Comment by clrux** [comment]
@talbs I've addressed the remaining bits of feedback. Let me know if I'm okay to merge.

File: _src/pattern-library/sass/base/_reset.scss [hunk 4]
```diff
@@ -105,6 +105,11 @@ pre {
 // ----------------------------
 .sr-only {
     @extend %a11y-hide;
+    @extend %a11y-ensure-contrast;
```
Comment (talbs) [inline _src/pattern-library/sass/base/_reset.scss]: Nice!

**Comment by talbs** [comment]
:+1:

**Pull request merged**